client/services/FeedbackGeneratorService.js

```diff
@@ -24,12 +24,76 @@ tie.factory('FeedbackGeneratorService', [
   'RUNTIME_ERROR_FEEDBACK_MESSAGES', 'WRONG_LANGUAGE_ERRORS', 'LANGUAGE_PYTHON',
   'CLASS_NAME_AUXILIARY_CODE', 'CLASS_NAME_SYSTEM_CODE', 'PARAGRAPH_TYPE_TEXT',
   'PARAGRAPH_TYPE_CODE', 'PARAGRAPH_TYPE_SYNTAX_ERROR',
+  'PYTHON_PRIMER_BUTTON_NAME',
   function(
     FeedbackObjectFactory, TranscriptService, ReinforcementGeneratorService,
     CODE_EXECUTION_TIMEOUT_SECONDS, SUPPORTED_PYTHON_LIBS,
     RUNTIME_ERROR_FEEDBACK_MESSAGES, WRONG_LANGUAGE_ERRORS, LANGUAGE_PYTHON,
     CLASS_NAME_AUXILIARY_CODE, CLASS_NAME_SYSTEM_CODE, PARAGRAPH_TYPE_TEXT,
-    PARAGRAPH_TYPE_CODE, PARAGRAPH_TYPE_SYNTAX_ERROR) {
+    PARAGRAPH_TYPE_CODE, PARAGRAPH_TYPE_SYNTAX_ERROR,
+    PYTHON_PRIMER_BUTTON_NAME) {
+
+    /**
+     * Constant for the number of times that a user can make a mistake (i.e.
+     * same error, syntax error, etc.) until we prompt them to look at the
+     * primer.
+     *
+     * @type {number}
+     */
+    var UNFAMILIARITY_THRESHOLD = 5;
+
+    /**
+     * Counter to keep track of consecutive syntax errors.
+     *
+     * @type {number}
+     */
+    var consecutiveSyntaxErrorCounter = 0;
+
+    /**
+     * Counter to keep track of consecutive same errors.
+     *
+     * @type {number}
+     */
+    var consecutiveSameRuntimeErrorCounter = 0;
+
+    /**
+     * Counter to keep track of consecutive uses of the wrong language.
+     *
+     * @type {number}
+     */
+    var consecutiveWrongLanguageErrorCounter = 0;
+
+    /**
+     * Variable to store the error string immediately before the current error.
+     * Will be used to see if the user is receiving the same exact error
+     * consecutively, a possible indication of language unfamiliarity.
+     *
+     * @type {string}
+     */
+    var previousErrorString = '';
+
+    /**
+     * Resets all but one specific counter or all counters, depending on
+     * the parameter.
+     *
+     * @param {string} currentCounter
+     * @private
+     */
+    var _resetCounters = function(currentCounter) {
+      // If the parameter is null, reset all counters.
+      var counterNotToReset = currentCounter || '';
+      if (counterNotToReset !== 'consecutiveSyntaxError') {
```

Comment:
I think it might have gotten lost in the outdated comments, but can we make these constants? Since we use them here and also in the other files when we call this method.